### ISSUE TITLE
Issue regarding the last step of self attention (weighted sum step)

### ISSUE THREAD

**Issue by pangzss** [ok1zjf/VASNet#25]
Hi, I noticed that the last step of the self-attention calculation doesn't seem so right:
```
att_weights_ = nn.functional.softmax(logits, dim=-1)       
weights = self.dropout(att_weights_)     
y = torch.matmul(V.transpose(1,0), weights).transpose(1,0)
```
So here the softmax probability is calculated along the dim -1, which is the column direction.
But then the weighted sum is taken along the row direction according to this line
```
y = torch.matmul(V.transpose(1,0), weights).transpose(1,0)
```
I think we should do something like this
```
y = torch.matmul(weights,V)
```
How do you think?
I hope I'm the one to be corrected.
